src/utils.js

```diff
@@ -0,0 +1,1320 @@
+  if (flag.length % 2 === 0) {
+    let i
+    const tempObj = {}
+    for (i = 0; i < flag.length; i += 2) {
+      try {
+        // assume it is JSON, there is only 1 way to find out
+        tempObj[flag[i]] = JSON.parse(flag[i + 1])
+      } catch (ex) {
+        // hmm ... not json, treat as string
+        tempObj[flag[i]] = flag[i + 1]
+      }
+    }
+    return tempObj
+  } else {
+    throw (new Error('Please provide correct values for flags'))
+  }
+}
+
+/**
+ * @description parses a string and returns the namespace and entity name for a package.
+ * @returns An object { namespace: string, name: string }
+ * @param name
+ */
+function parsePackageName (name) {
+  const delimiter = '/'
+  const parts = name.split(delimiter)
+  let n = parts.length
+  const leadingSlash = name[0] === delimiter
+  // accept no more than [/]ns/p
+  // these are all valid entries [/]ns/p, p, [/]_/p
+  if (n < 1 || n > 3 || (leadingSlash && n === 2) || (!leadingSlash && n === 3)) throw (new Error('Package name is not valid'))
+  // skip leading slash, all parts must be non empty (could tighten this check to match EntityName regex)
+  parts.forEach(function (part, i) { if (i > 0 && part.trim().length === 0) throw (new Error('Package name is not valid')) })
+  if (leadingSlash) {
+    parts.shift() // drop leading slash
+    n--
+  }
+  return {
+    namespace: n === 2 ? parts[0] : '_',
+    name: n === 1 ? parts[0] : parts[1]
+  }
+}
+
+/**
+ * @description returns key value pairs from the parameters supplied. Used to create --param-file and --annotation-file key value pairs
+ * @param file : flags['param-file'] or flags['annotation-file']
+ * @returns An object of key value pairs in this format : {Your key1 : 'Your Value 1' , Your key2: 'Your value 2'}
+ */
+function createKeyValueObjectFromFile (file) {
+  const jsonData = fs.readFileSync(file)
+  return JSON.parse(jsonData)
+}
+
+/**
+ * @param sequenceAction
+ */
+function createComponentsfromSequence (sequenceAction) {
+  const fqn = require('openwhisk-fqn')
+  const objSequence = {}
+  objSequence.kind = 'sequence'
+  // The components array requires fully qualified names [/namespace/package_name/action_name] of all the actions passed as sequence
+  objSequence.components = sequenceAction.map(component => {
+    return fqn(component)
+  })
+  return objSequence
+}
+
+/**
+ * @param firstObject
+ * @param secondObject
+ */
+function returnUnion (firstObject, secondObject) {
+  return Object.assign(firstObject, secondObject)
+}
+
+/**
+ * @param path
+ */
+function parsePathPattern (path) {
+  const pattern = /^\/(.+)\/(.+)$/i
+  const defaultMatch = [null, null, path]
+
+  return (pattern.exec(path) || defaultMatch)
+}
+
+/**
+ * @param input
+ * @param params
+ */
+function processInputs (input, params) {
+  // check if the value of a key is an object (Advanced parameters)
+  const dictDataTypes = {
+    string: '',
+    integer: 0,
+    number: 0
+  }
+
+  const output = Object.assign({}, input)
```

Comment:
I think we should use clonedeep from lodash here (since it's already a dep).

Object.assign is a shallow copy, so objects are copied by reference so you could still modify the input in the new object.